### ISSUE TITLE
Issues faced migrating ACER implementation to TF2

### ISSUE THREAD

**Issue by ghost** [openai/baselines#1163]
Note: this issue may not be directly related to baselines however I'm willing to contribute ACER to the `tf2` branch if someone is interested and can help me.

I'm creating implementations of my own relying on your implementations as a reference. I'm currently working on ACER implementation which is not available in `tf2` branch. I'm stuck in converting the trust region block to its equivalent in TF2.
Please check [acer.py](https://github.com/openai/baselines/blob/master/baselines/acer/acer.py) lines [155-176].
This is the part I'm interested in:

```
g = tf.gradients(- (loss_policy - ent_coef * entropy) * nsteps * nenvs, f) #[nenvs * nsteps, nact]
# k = tf.gradients(KL(f_pol || f), f)
k = - f_pol / (f + eps) #[nenvs * nsteps, nact] # Directly computed gradient of KL divergence wrt f
k_dot_g = tf.reduce_sum(k * g, axis=-1)
adj = tf.maximum(0.0, (tf.reduce_sum(k * g, axis=-1) - delta) / (tf.reduce_sum(tf.square(k), axis=-1) + eps)) #[nenvs * nsteps]

# Calculate stats (before doing adjustment) for logging.
avg_norm_k = avg_norm(k)
avg_norm_g = avg_norm(g)
avg_norm_k_dot_g = tf.reduce_mean(tf.abs(k_dot_g))
avg_norm_adj = tf.reduce_mean(tf.abs(adj))

g = g - tf.reshape(adj, [nenvs * nsteps, 1]) * k
grads_f = -g/(nenvs*nsteps) # These are turst region adjusted gradients wrt f ie statistics of policy pi
grads_policy = tf.gradients(f, params, grads_f)
grads_q = tf.gradients(loss_q * q_coef, params)
grads = [gradient_add(g1, g2, param) for (g1, g2, param) in zip(grads_policy, grads_q, params)]

avg_norm_grads_f = avg_norm(grads_f) * (nsteps * nenvs)
norm_grads_q = tf.global_norm(grads_q)
norm_grads_policy = tf.global_norm(grads_policy)
```

What I'm trying to do is to convert `tf.gradients` to `tf.GradientTape.gradients()` equivalent. Here's a simplified version of what I'm trying to do:
```
with tf.GradientTape() as tape:
    f = calculate_f()
    f_pol = calculate_f_pol()
    others = do_further_calculations()
    loss = calculate_loss()
g = tape.gradient(-loss, f)
```
However, I keep getting `[None]` as the output for `tape.gradient(-loss, f)` I created an [issue](https://github.com/tensorflow/tensorflow/issues/47387) in tensorflow as I'm not sure whether the problem is tf-specific vs implementation specific. I'd really appreciate if someone can help me figure out what's wrong and fix the issue. Feel free to ask any questions.
